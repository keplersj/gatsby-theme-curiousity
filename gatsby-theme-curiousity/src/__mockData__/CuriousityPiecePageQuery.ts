import { CuriousityPiecePage } from "./CuriousityPiecePage";
import { PiecePageQuery } from "../templates/piece-query";

export const CuriousityPiecePageQuery: PiecePageQuery = {
  portfolioItem: CuriousityPiecePage
};
