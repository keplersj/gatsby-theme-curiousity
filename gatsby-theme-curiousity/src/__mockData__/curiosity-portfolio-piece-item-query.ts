import { PortfolioPieceItemQuery } from "../components/PortfolioListItem";

export const CuriosityPortfolioPieceItemQuery: PortfolioPieceItemQuery = {
  site: {
    siteMetadata: {
      siteUrl: "https://example.dev/"
    }
  }
};
