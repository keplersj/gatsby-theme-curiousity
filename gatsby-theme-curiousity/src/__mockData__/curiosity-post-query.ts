import { PostQuery } from "../components/Piece";

export const CuriosityPostQuery: PostQuery = {
  site: {
    siteMetadata: {
      siteUrl: "https://example.dev/",
    },
  },
};
