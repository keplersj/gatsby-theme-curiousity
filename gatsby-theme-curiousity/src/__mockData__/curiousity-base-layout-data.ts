import { BaseLayoutData } from "../components/Base";

export const CuriousityBaseLayoutData: BaseLayoutData = {
  site: {
    siteMetadata: {
      title: "Test Site",
      siteUrl: "https://example.dev/"
    }
  }
};
