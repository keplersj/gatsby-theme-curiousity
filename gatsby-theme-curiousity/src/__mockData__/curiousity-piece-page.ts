import { PiecePage } from "../components/Piece";

export const CuriousityPiecePage: PiecePage = {
  id: "42",
  excerpt: "test excerpt",
  html: "<div><span>Test Piece</span></div>",
  slug: "/test-piece",
  title: "Test Piece",
  tags: [],
  keywords: [],
  metadata: {
    type: ["Test"],
    status: ["Test"],
    role: ["Test"],
    homepage: "https://example.dev/",
    rubygemsGemName: "test",
    npmPackageName: "test",
    githubRepo: "test/test",
  },
  fileRelativePath: "",
  rawFrontmatter: "",
  rawMarkdownBody: "",
  frontmatter: {},
};
