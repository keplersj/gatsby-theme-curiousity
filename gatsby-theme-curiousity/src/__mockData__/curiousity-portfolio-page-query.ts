import { PortfolioPageQuery } from "../templates/portfolio-query";

export const CuriousityPortfolioPageQuery: PortfolioPageQuery = {
  allPortfolioItem: {
    edges: [{ node: { id: "", excerpt: "", slug: "", title: "" } }],
  },
};
