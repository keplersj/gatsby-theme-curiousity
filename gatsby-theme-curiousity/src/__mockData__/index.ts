export * from "./CuriosityPortfolioPieceItemQuery";
export * from "./CuriosityPostQuery";
export * from "./CuriousityBaseLayoutData";
export * from "./CuriousityPiecePage";
export * from "./CuriousityPiecePageQuery";
export * from "./CuriousityPortfolioPageQuery";
