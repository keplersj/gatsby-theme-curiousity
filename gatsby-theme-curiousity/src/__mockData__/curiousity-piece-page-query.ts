import { CuriousityPiecePage } from "./curiousity-piece-page";
import { PiecePageQuery } from "../templates/piece-query";

export const CuriousityPiecePageQuery: PiecePageQuery = {
  portfolioItem: CuriousityPiecePage,
};
