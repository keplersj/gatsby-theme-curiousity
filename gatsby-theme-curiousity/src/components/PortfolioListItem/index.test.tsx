import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import { PortfolioListItem } from "./";
import { CuriosityPortfolioPieceItemQuery } from "../../__mockData__";

describe("Portfolio List Item Component", () => {
  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockImplementation(
      (): object => CuriosityPortfolioPieceItemQuery
    );
  });

  it("renders as expected", () => {
    const tree = renderer
      .create(
        <PortfolioListItem
          location="/item"
          title="Sample Portfolio Item"
          description="This is an example of a portfolio piece."
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
