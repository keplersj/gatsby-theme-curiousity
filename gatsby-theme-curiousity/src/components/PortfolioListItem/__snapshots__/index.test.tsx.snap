// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio List Item Component renders as expected 1`] = `
.emotion-2 {
  margin: 1em;
  background-color: rgba(var(--starstuff-card-default-background-color),0.5);
  border-radius: 10px;
  box-shadow: 0.5em 0.5em 0.75em rgba(var(--starstuff-card-default-shadow-color),0.3);
  -webkit-transition: box-shadow 1s;
  transition: box-shadow 1s;
  --starstuff-card-default-background-color: 245,245,245;
  --starstuff-card-default-shadow-color: 10,10,10;
}

.emotion-2:active,
.emotion-2:focus,
.emotion-2:focus-within,
.emotion-2:hover {
  box-shadow: 0.8em 0.8em 1em rgba(var(--starstuff-card-default-shadow-color),0.4);
}

@media screen and (prefers-color-scheme:dark) {
  .emotion-2 {
    --starstuff-card-default-background-color: 55,55,55;
  }
}

@media screen and (min-width:1024px) {
  .emotion-2 {
    min-height: 192px;
  }
}

@media screen and (max-width:512px) {
  .emotion-2 {
    max-width: 100vw;
  }
}

@media screen and (min-width:768px) {
  .emotion-2 {
    max-width: 33vw;
  }
}

@media screen and (min-width:1024px) {
  .emotion-2 {
    max-width: 25vw;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="emotion-2"
>
  <article
    className="emotion-1"
  >
    <script
      type="application/ld+json"
    >
      Object {
      "@context": "https://schema.org",
      "@id": "https://example.dev//item",
      "@type": "CreativeWork",
      "headline": "Sample Portfolio Item",
      "mainEntityOfPage": "https://example.dev//item",
      "name": "Sample Portfolio Item",
      "url": "https://example.dev//item",
    }
    </script>
    <div
      className="emotion-0"
    >
      <a
        href="/item"
      >
        <h2>
          Sample Portfolio Item
        </h2>
      </a>
      <p>
        This is an example of a portfolio piece.
      </p>
    </div>
  </article>
</div>
`;
